### DISCUSSION TITLE
[Deploy] Blue green deploy strategy

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#276] [Q&A] [answered]
### Which section?

Console

### Which is the version?

v11

### Description

Hello guys,
a client asks us to implement a blue/green deployment strategy.
We are trying to implement it using the console, the pipeline and customize only, with some manual action.

Did you ever implement this strategy without using additional tools?

Thanks.

**Comment by davidebianchi** [comment]
Hi! Interesting!
How do you think to implement it with kustomize? Create another namespace and route the api to the correct one? Or the blue-green of only one service?

We in the past have used the canary, both of a single service using the service Kubernetes than of the entire namespace routing with an api-gateway to release incrementally a feature.

We saw also to use additional tools, but we have never tried it in production.

**Reply by frengooo** [reply]
Hi @davidebianchi,
thanks for your response.

I don't really know, I am thinking about two strategies using the Console (e.g. duplicate the microservice to be released with the blue/green strategy), kustomize to change the service match labels to point to the right version.

I think with labels/annotations and the resources we have we can implement this strategy, with some manual action.

I'll let you know.

In the meantime, if you have some suggestion/proposal/other let me know 😄

**Reply by frengooo** [reply, marked as answer]
Hi guys,
we didn't implement the blue/green strategy using kustomize at the moment.

With [Enhanced Workflow](https://docs.mia-platform.eu/docs/development_suite/set-up-infrastructure/enhanced-project-workflow) and pull based approach, [with tools like ArgoCD/FluxCD](https://docs.mia-platform.eu/docs/infrastructure/self-hosted/third-party-integrations/deployment-tools) is possible to do that without custom flows.
